cardano-cli/src/Cardano/CLI/Shelley/Run/Address.hs

```diff
@@ -56,41 +54,66 @@ runAddressCmd cmd =
-runAddressKeyHash (VerificationKeyFile vkeyPath) mOutputFp =
-    firstExceptT ShelleyAddressCmdPayVerificationKeyReadErrr $ do
-      PaymentVerificationKeyShelley vkey <- ExceptT $ readPaymentVerificationKey vkeyPath
-      let Ledger.KeyHash khash = Ledger.hashKey vkey
-          b16Hash = Crypto.getHashBytesAsHex khash
-      case mOutputFp of
-        Just (OutputFile fpath) -> liftIO $ BS.writeFile fpath b16Hash
-        Nothing -> liftIO $ BS.putStrLn b16Hash
+runAddressKeyHash (VerificationKeyFile vkeyPath) mOutputFp = do
+  paymentVerKey <- firstExceptT ShelleyAddressCmdReadFileError
+    . newExceptT
+    $ readFileTextEnvelope (AsVerificationKey AsPaymentKey) vkeyPath
+
+  let hexKeyHash = serialiseToRawBytesHex (verificationKeyHash paymentVerKey)
+
+  case mOutputFp of
+    Just (OutputFile fpath) -> liftIO $ BS.writeFile fpath hexKeyHash
+    Nothing -> liftIO $ BS.putStrLn hexKeyHash
 
 runAddressBuild :: VerificationKeyFile
                 -> Maybe VerificationKeyFile
                 -> Network
                 -> Maybe OutputFile
                 -> ExceptT ShelleyAddressCmdError IO ()
-runAddressBuild (VerificationKeyFile payVkeyFp) mstkVkeyFp nw mOutFp =
-  firstExceptT ShelleyAddressCmdPayVerificationKeyReadErrr $ do
-    payVKey <- newExceptT $ readPaymentVerificationKey payVkeyFp
-    mstkVKey <- case mstkVkeyFp of
-                  Just (VerificationKeyFile stkVkeyFp) ->
-                    Just <$> newExceptT (readStakingVerificationKey stkVkeyFp)
-                  Nothing ->
-                    return Nothing
-    let addr = shelleyVerificationKeyAddress nw payVKey mstkVKey
-        hexAddr = addressToHex addr
+runAddressBuild (VerificationKeyFile payVkeyFp) mstkVkeyFp nw mOutFp = do
+    payVKey <- firstExceptT ShelleyAddressCmdReadFileError
+      . newExceptT
+      $ readFileTextEnvelope (AsVerificationKey AsPaymentKey) payVkeyFp
+    let paymentCred = PaymentCredentialByKey (verificationKeyHash payVKey)
+
+    stakeAddrRef <- firstExceptT ShelleyAddressCmdReadFileError $
+      case mstkVkeyFp of
+        Just (VerificationKeyFile stkVkeyFp) ->
+          toStakeAddrRef
+            <$> newExceptT (readFileTextEnvelope (AsVerificationKey AsStakeKey) stkVkeyFp)
+        Nothing -> pure NoStakeAddress
+
+    let addr = makeShelleyAddress nwId paymentCred stakeAddrRef
+        hexAddr = serialiseToRawBytesHex addr
+
     case mOutFp of
-      Just (OutputFile fpath) -> liftIO . BS.writeFile fpath $ textToByteString hexAddr
-      Nothing -> liftIO $ Text.putStrLn hexAddr
+      Just (OutputFile fpath) -> liftIO . BS.writeFile fpath $ hexAddr
+      Nothing -> liftIO $ BS.putStrLn hexAddr
+  where
+    toStakeAddrRef :: VerificationKey StakeKey -> StakeAddressReference
+    toStakeAddrRef = StakeAddressByValue . StakeCredentialByKey . verificationKeyHash
+
+    -- TODO: Remove this once we remove usage of 'Cardano.Api.Types.Network'
+    --       from this module.
+    nwId :: Api.NetworkId
```

Comment:
Should we put these conversion functions in their own module?

Reply:
I figured it probably wasn't worth it since this one in particular will be removed once we add a CLI parsers for the new API's `NetworkId` type.